jmx-scraper/src/main/java/io/opentelemetry/contrib/jmxscraper/client/JmxRemoteClient.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.jmxscraper.client;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.security.Provider;
+import java.security.Security;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.callback.NameCallback;
+import javax.security.auth.callback.PasswordCallback;
+import javax.security.auth.callback.UnsupportedCallbackException;
+import javax.security.sasl.RealmCallback;
+
+public class JmxRemoteClient {
+
+  private static final Logger logger = Logger.getLogger(JmxRemoteClient.class.getName());
+
+  private final JMXServiceURL url;
+  @Nullable private String userName;
+  @Nullable private String password;
+  @Nullable private String profile;
+  @Nullable private String realm;
+  private boolean sslRegistry;
+
+  private JmxRemoteClient(JMXServiceURL url) {
+    this.url = url;
+  }
+
+  public static JmxRemoteClient createNew(String host, int port) {
+    return new JmxRemoteClient(buildUrl(host, port));
```

Comment:
VERY stupid nit, but prefer chaining rather than duplicating calls to the constructors:

```suggestion
    return new createNew(buildUrl(host, port));
```